soda/core/soda/execution/data_source.py

```diff
@@ -205,33 +205,73 @@ def sql_select_all(self, table_name: str, limit: int | None = None) -> str:
     # For a table, get the columns metadata
     ############################################
 
-    def get_table_columns(self, table_name: str, query_name: str) -> dict[str, str]:
+    def get_table_columns(
+        self,
+        table_name: str,
+        query_name: str,
+        included_columns: list[str] | None = None,
+        excluded_columns: list[str] | None = None,
+    ) -> dict[str, str] | None:
         """
         :return: A dict mapping column names to data source data types.  Like eg
         {"id": "varchar", "size": "int8", ...}
         """
         query = Query(
             data_source_scan=self.data_source_scan,
             unqualified_query_name=query_name,
-            sql=self.sql_get_table_columns(table_name),
+            sql=self.sql_get_table_columns(
+                table_name, included_columns=included_columns, excluded_columns=excluded_columns
+            ),
         )
         query.execute()
-        return {row[0]: row[1] for row in query.rows}
+        if len(query.rows) > 0:
+            return {row[0]: row[1] for row in query.rows}
+        return None
 
     def create_table_columns_query(self, partition: Partition, schema_metric: SchemaMetric) -> TableColumnsQuery:
         return TableColumnsQuery(partition, schema_metric)
 
-    def sql_get_table_columns(self, table_name: str) -> str:
+    def sql_get_table_columns(
+        self, table_name: str, included_columns: list[str] | None = None, excluded_columns: list[str] | None = None
+    ) -> str:
+        # build optional filter clauses
+        if self.database:
+            database_filter = f" \n  AND lower({self.column_metadata_catalog_column()}) = '{self.database.lower()}'"
+        else:
+            database_filter = ""
+
+        if self.schema:
+            schema_filter = f" \n  AND lower(table_schema) = '{self.schema.lower()}'"
+        else:
+            schema_filter = ""
+
+        order_clause = f"\nORDER BY ORDINAL_POSITION"
```

Comment:
1) Why no do this inline in the `# compose  query template` below ?

2) What does it do and why is it added to the query?

Reply:
What does "this" refer to? You mean why not always add the `order by` statement in the template?

I don't have a problem with that. It just that previously in the old code, it was separate and I tend to try to leave original decisions mostly untouched unless they make things totally complicated or I see a strong problem with it. I'll take care of that now though

Reply:
Let me rephrase the questions:

1) Why is the `ORDER BY ORDINAL_POSITION` first assigned as a variable and later added in the template as a variable.  The variable only is assigned once and not changed.  So you could just add the text `ORDER BY ORDINAL_POSITION` straight  into the template below without using a variable

2) Explain what `ORDER BY ORDINAL_POSITION` does and why it is included in the query (if you know) as I don't know what it does and want to learn

Reply:
In the most recent code I've now made directly part of the string.

As for 2. I have no idea, it was there before, and I didn't question it. However, I propose to create a follow up ticket to investigate its meaning and whether it's required at all so that if it's not we can remove it. Would that be a good idea?